eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Procedures.hs

```diff
@@ -296,13 +304,66 @@ instance EraPParams era => EncCBOR (ProposalProcedure era) where
         !> To pProcGovernanceAction
         !> To pProcAnchor
 
+data Constitution era = Constitution
+  { constitutionHash :: !(SafeHash (EraCrypto era) ByteString)
+  , constitutionScript :: !(StrictMaybe (ScriptHash (EraCrypto era)))
+  }
+  deriving (Generic)
+
+constitutionHashL :: Lens' (Constitution era) (SafeHash (EraCrypto era) ByteString)
+constitutionHashL = lens constitutionHash (\x y -> x {constitutionHash = y})
+
+constitutionScriptL :: Lens' (Constitution era) (StrictMaybe (ScriptHash (EraCrypto era)))
+constitutionScriptL = lens constitutionScript (\x y -> x {constitutionScript = y})
+
+instance Era era => ToJSON (Constitution era) where
+  toJSON = object . toConstitutionPairs
+  toEncoding = pairs . mconcat . toConstitutionPairs
+
+toConstitutionPairs :: (KeyValue a, Era era) => Constitution era -> [a]
+toConstitutionPairs Constitution {..} =
+  [ "constitutionHash" .= constitutionHash
+  , "constitutionScript" .= constitutionScript
+  ]
```

Comment:
It might be better to omit the script hash if is not present
```suggestion
  ] ++ [ "constitutionScript" .= script | SJust script <- constitutionScript]
```